### PR TITLE
New one to one conversation creation flow

### PR DESCRIPTION
**Pull request by Ivansss**
| Conversation list | 1. Create a new conversation <br> 2. Join open conversation <br> 3. Create one2one conversation  | Create a new conversation  |
|---|---|---|
| IMG_E0049 | IMG_E0046 | IMG_E0048 |

### CODE REVIEW

File: NextcloudTalk/NewRoomTableViewController.swift [hunk 1]
```diff
@@ -0,0 +1,303 @@
...
+    case joinOpenRoom
+}
+
+@objcMembers class NewRoomTableViewController: UITableViewController,
+                                               UINavigationControllerDelegate,
+                                               UISearchResultsUpdating {
+
+    var account: TalkAccount
+
+    var indexes: [String] = [""]
```
Comment (SystemKeeper) [inline NextcloudTalk/NewRoomTableViewController.swift:35]: This feels wrong 🤔 Is it to align the section number to the index?

File: NextcloudTalk/NewRoomTableViewController.swift [hunk 2]
```diff
@@ -0,0 +1,303 @@
...
+        super.viewWillAppear(animated)
+
+        self.navigationItem.hidesSearchBarWhenScrolling = false
+        self.getPossibleContacts()
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        self.navigationController?.navigationBar.titleTextAttributes = [NSAttributedString.Key.foregroundColor: NCAppBranding.themeTextColor()]
```
Comment (SystemKeeper) [inline NextcloudTalk/NewRoomTableViewController.swift]: Could use https://github.com/nextcloud/talk-ios/blob/master/NextcloudTalk/NCAppBranding.m#L228 until we have a super class for this.

File: NextcloudTalk/NewRoomTableViewController.swift [hunk 3]
```diff
@@ -0,0 +1,303 @@
...
+            }
+        }
+    }
+
+    // MARK: - Search
+
+    func updateSearchResults(for searchController: UISearchController) {
+        self.resultTableViewController.showSearchingUI()
+        DispatchQueue.main.async {
+            self.searchTimer = Timer.scheduledTimer(timeInterval: 1.0, target: self, selector: #selector(self.searchForContacts), userInfo: nil, repeats: false)
```
Comment (SystemKeeper) [inline NextcloudTalk/NewRoomTableViewController.swift:151]: I think there's a missing invalidation here. Shouldn't the search timer be invalidated at some point to have a correct debounce?

File: NextcloudTalk/NewRoomTableViewController.swift [hunk 4]
```diff
@@ -0,0 +1,303 @@
...
+        let navController = NCNavigationController(rootViewController: viewController)
+        self.present(navController, animated: true)
+    }
+
+    func createRoomWithContact(_ contact: NCUser) {
+        NCAPIController.sharedInstance().createRoom(forAccount: account, withInvite: contact.userId, ofType: .oneToOne, andName: nil) { room, error in
+            if let token = room?.token, error == nil {
+                self.navigationController?.dismiss(animated: true, completion: {
+                    NotificationCenter.default.post(name: NSNotification.Name.NCSelectedUserForChat, object: self, userInfo: ["token": token])
+                })
```
Comment (SystemKeeper) [inline NextcloudTalk/NewRoomTableViewController.swift]: Could also write it like: 

```suggestion
                self.navigationController?.dismiss(animated: true) {
                    NotificationCenter.default.post(name: NSNotification.Name.NCSelectedUserForChat, object: self, userInfo: ["token": token])
                }
```

File: NextcloudTalk/NewRoomTableViewController.swift (hunk 1 above)
Comment (Ivansss) [reply NextcloudTalk/NewRoomTableViewController.swift:35]: This is basically the section we'll use for the "Create a new conversation" and "Join open conversations" options.
We can take another (more self-explainatory) approach sure ;)

File: NextcloudTalk/NewRoomTableViewController.swift (hunk 2 above)
Comment (Ivansss) [reply NextcloudTalk/NewRoomTableViewController.swift]: Forgot about that function. I agree, we should use it until we create a super class 👍

File: NextcloudTalk/NewRoomTableViewController.swift (hunk 3 above)
Comment (Ivansss) [reply NextcloudTalk/NewRoomTableViewController.swift:151]: I was pretty sure I added the timer invalidation here :P

File: NextcloudTalk/NewRoomTableViewController.swift (hunk 2 above)
Comment (Ivansss) [reply NextcloudTalk/NewRoomTableViewController.swift]: Fixed!

File: NextcloudTalk/NewRoomTableViewController.swift (hunk 3 above)
Comment (Ivansss) [reply NextcloudTalk/NewRoomTableViewController.swift:151]: Fixed!

File: NextcloudTalk/ContactsSearchResultTableViewContoller.swift [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+//
+
+import UIKit
+
+@objcMembers class ContactsSearchResultTableViewController: UITableViewController {
+
+    var indexes: [String] = []
+    var contacts: [String: [NCUser]] = [:]
+
+    let tableBackgroundView = PlaceholderView()
```
Comment (SystemKeeper) [inline NextcloudTalk/ContactsSearchResultTableViewContoller.swift]:
```suggestion
    let tableBackgroundView = PlaceholderView(for: .insetGrouped)!
```


https://github.com/nextcloud/talk-ios/assets/1580193/a1a31f82-9e69-4310-b6ff-c07e83ba73ed

**Review by SystemKeeper — APPROVED** [review]
One small design nit-pick, otherwise really nice job 👌

File: NextcloudTalk/NewRoomTableViewController.swift (hunk 1 above)
Comment (SystemKeeper) [reply NextcloudTalk/NewRoomTableViewController.swift:35]: Not really sure how to solve it easily, without counting differently in all other methods. I'd say we keep it for now like this?!

**Pull request merged**